Storage/Bookmarks/Bookmarks.swift

```diff
@@ -22,6 +22,15 @@ public protocol SyncableBookmarks: class, ResettableSyncStorage, AccountRemovalD
     func applyLocalOverrideCompletionOp(op: LocalOverrideCompletionOp, itemSources: ItemSources) -> Success
 }
 
+public enum BookmarkBufferState {
+    case NoUnmergedRemoteChanges
+    case UnmergedRemoteChanges
+}
+
+public protocol BufferedBookmarks: class {
+    func bufferState() -> Deferred<Maybe<BookmarkBufferState>>
```

Comment:
I don't see you call this anywhere in this PR. Am I missing something?

Even so, I question whether this needs to be a separate protocol. This should be hidden within the factory source, no?